falco/client.py

```diff
@@ -1,23 +1,46 @@
 import grpc
 
 from falco.client_credentials import get_grpc_channel_credentials
-from falco.domain import Response
-from falco.svc.output_pb2_grpc import serviceStub
+from falco.domain import Response, Request
+from falco.svc.outputs_pb2_grpc import serviceStub
+from falco.svc.version_pb2_grpc import serviceStub as versionServiceStub
+from falco.schema.version_pb2 import request as versionRequest
+from falco.schema.outputs_pb2 import request as outputsRequest
 
 
 class InvalidFormat(Exception):
     pass
 
+class TLSConfigError(Exception):
+    pass
+
+class RequestGenerator:
+    def __init__(self):
+        pass
+
+    def EmptyRequests(self):
+        while True:
+            yield outputsRequest()
 
 class Client:
-    def __init__(self, endpoint, client_crt, client_key, ca_root, output_format=None, *args, **kw):
-        self._client = serviceStub(
-            grpc.secure_channel(
+    def __init__(self, endpoint, client_crt=None, client_key=None, ca_root=None, output_format=None, *args, **kw):
+        if endpoint.startswith("unix:///"):
+            channel = grpc.insecure_channel(
+                endpoint,
+                options=[("grpc.max_receive_message_length", 1024 * 1024 * 512)],
```

Comment:
Is this value (1024*1024*512) for receiving messages exact?

Reply:
This is copy and pasted from the tls connection below. I didn't find us setting a maximum size in the falco source code to match this with.